src/Admin/AjaxAction.php

```diff
@@ -0,0 +1,225 @@
+	 * @param int      $scope    Optional. Scope of where the AJAX handler is listening. Defaults to backend only.
+	 */
+	public function __construct(
+		$action,
+		callable $callback,
+		$selector = '',
+		$access = self::ACCESS_USER,
+		$scope = self::SCOPE_BACKEND
+	) {
+		if ( ! is_int( $access ) || $access < self::ACCESS_ANY || $access > self::ACCESS_NO_USER ) {
+			$access = self::ACCESS_USER;
+		}
+
+		if ( ! is_int( $scope ) || $scope < self::SCOPE_BOTH || $scope > self::SCOPE_FRONTEND ) {
+			$scope = self::SCOPE_BACKEND;
+		}
+
+		$this->action   = $action;
+		$this->callback = $callback;
+		$this->selector = $selector;
+		$this->access   = $access;
+		$this->scope    = $scope;
+	}
+
+	/**
+	 * Register the AJAX action with the WordPress system.
+	 */
+	public function register() {
+		if ( in_array( $this->access, [ self::ACCESS_USER, self::ACCESS_ANY ], true ) ) {
+			add_action( $this->get_action( self::ACCESS_USER ), $this->callback );
+		}
+
+		if ( in_array( $this->access, [ self::ACCESS_NO_USER, self::ACCESS_ANY ], true ) ) {
+			add_action( $this->get_action( self::ACCESS_NO_USER ), $this->callback );
+		}
+
+		if ( in_array( $this->scope, [ self::SCOPE_FRONTEND, self::SCOPE_BOTH ], true )
+			&& ! has_action( static::FRONTEND_ENQUEUE_ACTION, [ $this, 'register_ajax_script' ] ) ) {
+			add_action( static::FRONTEND_ENQUEUE_ACTION, [ $this, 'register_ajax_script' ] );
+		}
+
+		if ( in_array( $this->scope, [ self::SCOPE_BACKEND, self::SCOPE_BOTH ], true )
+			&& ! has_action( static::BACKEND_ENQUEUE_ACTION, [ $this, 'register_ajax_script' ] ) ) {
+			add_action( static::BACKEND_ENQUEUE_ACTION, [ $this, 'register_ajax_script' ] );
+		}
+	}
+
+	/**
+	 * Register the AJAX logic.
+	 */
+	public function register_ajax_script() {
+		if ( empty( $this->selector ) ) {
+			return;
+		}
+
+		if ( self::SCOPE_FRONTEND === $this->scope && is_admin() ) {
+			return;
+		}
+
+		if ( self::SCOPE_BACKEND === $this->scope && ! is_admin() ) {
+			return;
+		}
+
+		$selector = wp_json_encode( $this->selector );
+		$action   = wp_json_encode( $this->action );
+
+		$script = <<< JS_SCRIPT
+;( function () {
+    var selector = {$selector};
+    setTimeout( function () {
+        ( document.querySelectorAll( selector ) || [] )
+            .forEach( ( element ) => {
+                element.addEventListener( 'click', function ( event ) {
+                    event.preventDefault();
+                    wp.ajax.post( {$action}, {} )
+                        .done( function () {
+                            element.classList.remove( 'ajax-failure' );
+                            element.classList.add( 'ajax-success' )
+                        } )
+                        .fail( function () {
+                            element.classList.remove( 'ajax-success' );
+                            element.classList.add( 'ajax-failure' )
+                        } );
+                } );
+            } );
+    }, 1000 );
```

Comment:
Why the 1-second timeout? Wouldn't a `DOMContentLoaded` event handler suffice?

Reply:
I tried that but it seems that the SiteHealth screen does other magic _after_ the DOM content was loaded, and I also couldn't find an event to check for. Any suggestion what I could use here instead of a dumb timeout?

Reply:
Hmm, it seems to work now. I must have done something else wrong when I tried it in the beginning.